### PR TITLE
526 separation/training pay

### PR DESCRIPTION
**Pull request by bshyong**
## Description
Reorders sections/screens relating to separation and training pay

## Testing done
local testing

## Acceptance criteria
- [x] Sections are reordered and modified according to acceptance criteria in https://github.com/department-of-veterans-affairs/vets.gov-team/issues/15932

## Definition of done
- [x] Events are logged appropriately
- [x] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

Resolves https://github.com/department-of-veterans-affairs/vets.gov-team/issues/15932

### CODE REVIEW

File: src/applications/disability-benefits/all-claims/config/form.js [hunk 1]
```diff
@@ -40,53 +40,54 @@ import { veteranInfoDescription } from '../content/veteranDetails';
 import { disabilitiesOrientation } from '../content/disabilitiesOrientation';
 import { supportingEvidenceOrientation } from '../content/supportingEvidenceOrientation';
 import {
```
Comment (bshyong) [inline src/applications/disability-benefits/all-claims/config/form.js]: alphabetization

**Review by cvalarida — COMMENTED** [review]
I wasn't 100% sure what was going on in the ticket, so I asked @djmunoz1 for some clarification. Sounds like we've got one more requirement not outlined in that ticket.

@djmunoz1 When you get a chance, can you update the connected ticket to include the requirement for moving the retirement pay page and renaming it?

**Comment by markgreenburg** [comment]
LMK when the page order is updated per latest comments in attached ticket and will be happy to review!

**Comment by bshyong** [comment]
This has been updated:
- renamed `servicePay` to `retirementPay`
- reordered `retirementPay` per issue https://github.com/department-of-veterans-affairs/vets.gov-team/issues/15932#issuecomment-452083814
- renamed `waive-retirement-pay` to `retirement-pay-waiver`
- modified unit tests to account for renaming

**Review by markgreenburg — APPROVED** [review]
LGTM! Not sure if Chris's page index changes or this should be merged first.

**Pull request merged**